packages/core-data/src/resolvers.js

```diff
@@ -585,10 +585,14 @@ export const getNavigationFallbackId =
 		dispatch.receiveNavigationFallbackId( fallback?.id );
 
 		if ( record ) {
+			const invalidateNavigationQueries = true;
+
 			dispatch.receiveEntityRecords(
 				'postType',
 				'wp_navigation',
-				record
+				record,
+				undefined,
+				invalidateNavigationQueries
```

Comment:
Good idea to label this param.

Reply:
That seems a bit weird.. 🤔 . Have you tried using `shouldInvalidate` in the resolver?

Reply:
The query we care about invalidating is `getEntityRecords`. That will not invalidate unless `invalidateNavigationQueries` is passed in the action here. 

Maybe we've misunderstood how this is supposed to work?